### PR TITLE
Remove unnecessary calls to .obj on MLContact

### PR DESCRIPTION
**Pull request by matthewrfennell**
We must explicitly cast MLContact.account to type xmpp?, as Swift struggles to infer this by itself.

### CODE REVIEW

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]
why not use `contact.account as! xmpp`? That should work, too, no?

File: Monal/Classes/OmemoQrCodeView.swift [hunk 1]
```diff
@@ -30,7 +30,7 @@ struct OmemoQrCodeView: View {
     init(contact: ObservableKVOWrapper<MLContact>)
     {
         self.jid = contact.obj.contactJid
-        if let account = contact.obj.account {
+        if let account = contact.account as xmpp? {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/OmemoQrCodeView.swift:33]:
```suggestion
        if let account = contact.account as? xmpp {
```

File: Monal/Classes/OmemoKeys.swift [hunk 2]
```diff
@@ -292,7 +292,7 @@ struct OmemoKeys: View {
         self.viewContact = contact
 
         if let contact = contact {
-            if let account = contact.obj.account {
+            if let account = contact.account as xmpp? {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/OmemoKeys.swift:295]: same as below

File: Monal/Classes/MemberList.swift [hunk 3]
```diff
@@ -30,7 +30,7 @@ struct MemberList: View {
     @StateObject private var overlay = LoadingOverlayState()
 
     init(mucContact: ObservableKVOWrapper<MLContact>) {
-        account = mucContact.obj.account! as xmpp
+        account = (mucContact.account as xmpp?)!
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/MemberList.swift:33]: same as above

File: Monal/Classes/EditGroupSubject.swift [hunk 4]
```diff
@@ -19,7 +19,7 @@ struct EditGroupSubject: View {
         
         _subject = State(wrappedValue: contact.obj.groupSubject)
         _contact = StateObject(wrappedValue: contact)
-        self.account = contact.obj.account! as xmpp
+        self.account = (contact.account as xmpp?)!
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/EditGroupSubject.swift:22]: same as above

File: Monal/Classes/ContactDetails.swift [hunk 5]
```diff
@@ -35,7 +35,7 @@ struct ContactDetails: View {
     init(delegate: SheetDismisserProtocol, contact: ObservableKVOWrapper<MLContact>) {
         self.delegate = delegate
         _contact = StateObject(wrappedValue: contact)
-        self.account = contact.obj.account!
+        self.account = (contact.account as xmpp?)!
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ContactDetails.swift:38]: same as above

File: Monal/Classes/ChannelMemberList.swift [hunk 6]
```diff
@@ -15,7 +15,7 @@ struct ChannelMemberList: View {
     @State private var participants: OrderedDictionary<String, String>
 
     init(mucContact: ObservableKVOWrapper<MLContact>) {
-        account = mucContact.obj.account! as xmpp
+        account = (mucContact.account as xmpp?)!
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ChannelMemberList.swift:18]:
```suggestion
        account = mucContact.account as! xmpp
```

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]

**Comment by matthewrfennell** [comment]
> why not use `contact.account as! xmpp`? That should work, too, no?

I should've mentioned - I tried these but in each case I get "Generic parameter 'T' could not be inferred"

It gets confused because of the `_Nullable` specifier I put in `MLContact`. If I remove that, I can apply these changes, but I think we need the `_Nullable` since it can return `nil`.

**Comment by tmolitor-stud-tu** [comment]
Yes, we should keep the `_Nullable`...what about `as! xmpp?`?

**Comment by tmolitor-stud-tu** [comment]
Ah: and we should use an `if let` unless we can be 100% sure that the account can never be nil...

**Pull request merged**